lamindb/core/_mapped_collection.py

```diff
@@ -79,6 +79,7 @@ def __init__(
         unknown_label: Optional[Union[str, Dict[str, str]]] = None,
         cache_categories: bool = True,
         parallel: bool = False,
+        add_dataset_id: bool = False,
```

Comment:
@Koncopd - I don't fully understand what this does, but likely it deserves a better name.